python/lsst/pipe/base/configOverrides.py

```diff
@@ -25,15 +25,108 @@
 from enum import Enum
 
 OverrideTypes = Enum("OverrideTypes", "Value File Python Instrument")
 
 
+class ConfigExpressionParser(ast.NodeVisitor):
+    """An expression parser that will be used to transform configuration
+    strings supplied from the command line or a pipeline into a python
+    object.
+
+    This is roughly equivalent to ast.literal_parser, but with the ability to
+    transform strings that are valid variable names into the value associated
+    with the name. Variables that should be considered valid are supplied to
+    the constructor as a dictionary that maps a string to its corresponding
+    value.
+
+    This class in an internal implementation detail, and should not be exposed
+    outside this module.
+
+    Parameters
+    ----------
+    namespace : `dict` of `str` to variable
+    """
+
+    def __init__(self, namespace):
+        self.variables = namespace
+
+    def visit_Name(self, node):
+        """This method gets called when the parser has determined a node
+        corresponds to a variable name.
+        """
+        # If the id (name) of the variable is in the dictionary of valid names,
+        # load and return the corresponding variable.
+        if node.id in self.variables:
+            return self.variables[node.id]
+        # If the node does not correspond to a valid variable, turn the name
+        # into a string, as the user likely intended it as such.
+        return f"{node.id}"
+
+    def visit_List(self, node):
+        """This method is visited if the node is a list. Constructs a list out
+        of the sub nodes.
+        """
+        result = []
+        for elm in node.elts:
+            result.append(self.visit(elm))
+        return result
+
+    def visit_Tuple(self, node):
+        """This method is visited if the node is a tuple. Constructs a list out
+        of the sub nodes, and then turns it into a tuple.
+        """
+        return tuple(self.visit_List(node))
+
+    def visit_Constant(self, node):
+        """This method is visited if the node is a constant
+        """
+        return node.value
+
+    def visit_Dict(self, node):
+        """This method is visited if the node is a dict. It builds a dict out
+        of the component nodes.
+        """
+        result = {}
+
+        for key, value in zip(node.keys, node.values):
+            result[self.visit(key)] = self.visit(value)
+        return result
+
+    def visit_Set(self, node):
+        """This method is visited if the node is a set. It builds a set out
+        of the component nodes.
+        """
+        result = set()
+        for el in node.elts:
+            result.add(self.visit(el))
+        return result
+
+    def visit_UnaryOp(self, node):
+        """This method is visited if the node is a unary operator. Currently
+        The only operator we support is the negative (-) operator, all others
+        are passed to generic_visit method.
+        """
+        if isinstance(node.op, ast.USub):
```

Comment:
`UAdd` may also be trivial to support here?

Reply:
It is, I just didnt expect someone to write something like `+17.2`, though I guess for symmetry it might be nice?